src/diart/inference.py

```diff
@@ -97,18 +97,22 @@ def __init__(
 
         self.stream = self.source.stream
 
+        # Rearrange stream to form sliding windows
+        self.stream = self.stream.pipe(
+            dops.rearrange_audio_stream(chunk_duration, step_duration, source.sample_rate),
+        )
+
         # Dynamic resampling if the audio source isn't compatible
         if sample_rate != self.source.sample_rate:
             msg = f"Audio source has sample rate {self.source.sample_rate}, " \
                   f"but pipeline's is {sample_rate}. Will resample."
             logging.warning(msg)
             self.stream = self.stream.pipe(
-                ops.map(blocks.Resample(self.source.sample_rate, sample_rate))
+                ops.map(blocks.Resample(self.source.sample_rate, sample_rate, self.pipeline.config.device))
```

Comment:
Resample before `rearrange_audio_stream` so the same audio is not resampled multiple times.
Because of how the first 5s buffer is filled at the beginning, this actually means that `Resample` will be called more times, but (unless it's running on GPU) each call should also be faster because the size of the chunk is reduced by 10 (80k vs 8k samples)

Reply:
Let's track this for a future PR, see #180